devicetypes/sky-nie/min-smart-plug-dimmer.src/min-smart-plug-dimmer.groovy

```diff
@@ -0,0 +1,496 @@
+        capability "Actuator"
+        capability "Sensor"
+        capability "Switch"
+        capability "Temperature Measurement"
+        capability "Switch Level"
+        capability "Configuration"
+        capability "Refresh"
+
+        attribute "firmwareVersion", "string"
+        attribute "lastCheckIn", "string"
+        attribute "syncStatus", "string"
+
+        fingerprint mfr: "0312", prod: "FF00", model: "FF0D", deviceJoinName: "Minoston Dimmer Switch" //MP21ZD
+        fingerprint mfr: "0312", prod: "FF07", model: "FF03", deviceJoinName: "Minoston Dimmer Switch" //MP22ZD
+    }
+
+    preferences {
+        configParams.each {
+            if (it.range) {
+                input "configParam${it.num}", "number", title: "${it.name}:", required: false, defaultValue: "${it.value}", range: it.range
+            } else {
+                input "configParam${it.num}", "enum", title: "${it.name}:", required: false, defaultValue: "${it.value}", options: it.options
+            }
+        }
+    }
+}
+
+def installed() {
+    logDebug "installed()..."
+    sendEvent(name: "checkInterval", value: checkInterval, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+    state.refreshConfig = true
+}
+
+private static def getCheckInterval() {
+    // These are battery-powered devices, and it's not very critical
+    // to know whether they're online or not – 12 hrs
+    return (60 * 60 * 3) + (5 * 60)
+}
+
+def updated() {
+    if (!isDuplicateCommand(state.lastUpdated, 5000)) {
+        state.lastUpdated = new Date().time
+
+        logDebug "updated()..."
+        if (device.latestValue("checkInterval") != checkInterval) {
+            sendEvent(name: "checkInterval", value: checkInterval, displayed: false)
+        }
+
+        runIn(5, executeConfigureCmds, [overwrite: true])
+    }
+
+    return []
+}
+
+def configure() {
+    logDebug "configure()..."
+
+    if (state.resyncAll == null) {
+        state.resyncAll = true
+        runIn(8, executeConfigureCmds, [overwrite: true])
+    } else {
+        if (!pendingChanges) {
+            state.resyncAll = true
+        }
+        executeConfigureCmds()
+    }
+    return []
+}
+
+def executeConfigureCmds() {
+    runIn(6, refreshSyncStatus)
+
+    def cmds = []
+
+    configParams.each { param ->
+        def storedVal = getParamStoredValue(param.num)
+        def paramVal = param.value
+        if (state.resyncAll || ("${storedVal}" != "${paramVal}")) {
+            cmds << secureCmd(zwave.configurationV1.configurationSet(parameterNumber: param.num, size: param.size, scaledConfigurationValue: paramVal))
+            if (param.num == temperatureReportThresholdParam.num) {
+                cmds << "delay 3000"
+                cmds << secureCmd(zwave.sensorMultilevelV5.sensorMultilevelGet(scale: 0, sensorType: 1))
+            } else {
+                cmds << secureCmd(zwave.configurationV1.configurationGet(parameterNumber: param.num))
+            }
```

Comment:
because of this code, your temperature report threshold will never be marked as synced

Reply:
oh, I checked the history version, the bug is happened for my sync mistake.

Reply:
oh, I checked the history version, the bug is happened for my sync mistake.

Reply:
@KKlimczukS @greens I have updated [the PR](https://github.com/sky-nie/SmartThingsPublic/tree/devws-for-nie-tech-co-ltd-contai_48). Can you review it at a convenient time?